src/7-execsnoop/execsnoop.bpf.c

```diff
@@ -26,7 +26,8 @@ int tracepoint__syscalls__sys_enter_execve(struct trace_event_raw_sys_enter* ctx
 	event.uid = uid;
 	task = (struct task_struct*)bpf_get_current_task();
 	event.ppid = BPF_CORE_READ(task, real_parent, tgid);
-	bpf_get_current_comm(&event.comm, sizeof(event.comm));
```

Comment:
It's my mistake. It should be:

```
SEC("tracepoint/syscalls/sys_exit_execve")
int tracepoint__syscalls__sys_exit_execve(struct trace_event_raw_sys_enter* ctx)
```
Then we can use `bpf_get_current_comm` to get the command for subprocess.

You can refer to https://github.com/iovisor/bcc/blob/master/libbpf-tools/execsnoop.bpf.c for the original bpf code.

Using bpf_probe_read_str is OK as well.

Reply:
I tried the code you mentioned and it works.